contrib/kube-prometheus/hack/cluster-monitoring/deploy

```diff
@@ -32,6 +32,12 @@ kctl apply -f manifests/prometheus/prometheus-k8s-service.yaml
 kctl apply -f manifests/alertmanager/alertmanager-config.yaml
 kctl apply -f manifests/alertmanager/alertmanager-service.yaml
 
+# unfortunately statefulsets cannot be changed except for their replica count
+# so we need to make sure that the rule files are created before we create the
+# prometheus resource so it can properly discover the rule files when creating
+# the statefulset
+sleep 5
```

Comment:
Mh, that's the point of the sidecar though. Removing the dynamic configmap mounts from the pod template and making it all a runtime internal of the pod.

Currently extending selected configmaps would literally mean deleting the entire stateful set first, which is pretty disruptive.